pkg/operator/staticpodoperator/controlloops/installer_controller.go

```diff
@@ -0,0 +1,377 @@
+	failedAtPrev := currKubeletState.LastFailedDeploymentID == prevKubeletState.CurrentDeploymentID
+	if prevAhead && !failedAtPrev {
+		return currKubeletState.CurrentDeploymentID
+	}
+
+	return 0
+}
+
+func getInstallerPodName(deploymentID int32, nodeName string) string {
+	return fmt.Sprintf("installer-%d-%s", deploymentID, nodeName)
+}
+
+// createInstallerPod creates the installer pod with the secrets required to
+func (c *InstallerController) createInstallerPod(currKubeletState *KubeletState, operatorSpec *operatorv1alpha1.OperatorSpec, deploymentID int32) (*KubeletState, error) {
+	required := resourceread.ReadPodV1OrDie([]byte(installerPod))
+	switch corev1.PullPolicy(operatorSpec.ImagePullPolicy) {
+	case corev1.PullAlways, corev1.PullIfNotPresent, corev1.PullNever:
+		required.Spec.Containers[0].ImagePullPolicy = corev1.PullPolicy(operatorSpec.ImagePullPolicy)
+	case "":
+	default:
+		return nil, fmt.Errorf("invalid imagePullPolicy specified: %v", operatorSpec.ImagePullPolicy)
+	}
+	required.Name = getInstallerPodName(deploymentID, currKubeletState.NodeName)
+	required.Spec.NodeName = currKubeletState.NodeName
+	required.Spec.Containers[0].Image = getInstallerPodImage()
+	required.Spec.Containers[0].Args = append(required.Spec.Containers[0].Args,
+		fmt.Sprintf("-v=%d", operatorSpec.Logging.Level),
+		fmt.Sprintf("--deployment-id=%d", deploymentID),
+		fmt.Sprintf("--namespace=%s", c.targetNamespace),
+		fmt.Sprintf("--pod=%s", c.configMaps[0]),
+		fmt.Sprintf("--resource-dir=%s", "/etc/kubernetes/static-pod-resources"),
+		fmt.Sprintf("--pod-manifest-dir=%s", "/etc/kubernetes/manifests"),
+	)
+	for _, name := range c.configMaps {
+		required.Spec.Containers[0].Args = append(required.Spec.Containers[0].Args, fmt.Sprintf("--configmaps=%s", name))
+	}
+	for _, name := range c.secrets {
+		required.Spec.Containers[0].Args = append(required.Spec.Containers[0].Args, fmt.Sprintf("--secrets=%s", name))
+	}
+
+	if _, err := c.kubeClient.CoreV1().Pods(c.targetNamespace).Create(required); err != nil {
+		glog.Errorf("failed to create pod for %q: %v", currKubeletState.NodeName, resourceread.WritePodV1OrDie(required))
+		return nil, err
+	}
+
+	ret := currKubeletState.DeepCopy()
+	ret.TargetDeploymentID = deploymentID
+	ret.Errors = nil
+
+	return ret, nil
+}
+
+func getInstallerPodImage() string {
+	if val := os.Getenv("OPERATOR_IMAGE"); len(val) > 0 {
+		return val
+	}
+
+	return "docker.io/openshift/origin-cluster-kube-apiserver-operator:v4.0"
+}
+
+func (c InstallerController) sync() error {
+	operatorSpec, originalOperatorStatus, resourceVersion, err := c.operatorConfigClient.Get()
+	if err != nil {
+		return err
+	}
+	operatorStatus := originalOperatorStatus.DeepCopy()
+
+	switch operatorSpec.ManagementState {
+	case operatorv1alpha1.Unmanaged:
+		return nil
+	case operatorv1alpha1.Removed:
+		// TODO probably just fail.  Static pod managers can't be removed.
```

Comment:
yeah, or set the error condition about unsupported management state?

Reply:
> yeah, or set the error condition about unsupported management state?

future?